evalml/tests/component_tests/test_stacked_ensemble_classifier.py

```diff
@@ -4,169 +4,139 @@
-        "n_jobs": -1,
+def test_stacked_ensemble_does_not_overwrite_pipeline_random_seed():
+    component_graph = {
+        "Random Forest": [RandomForestClassifier(random_seed=3), "X", "y"],
+        "Random Forest B": [RandomForestClassifier(random_seed=4), "X", "y"],
+        "Stacked Ensemble": [
+            StackedEnsembleClassifier(
+                n_jobs=1, final_estimator=RandomForestClassifier()
+            ),
+            "Random Forest.x",
+            "Random Forest B.x",
+            "y",
+        ],
     }
-    assert clf.parameters == expected_parameters
-    clf.fit(X, y)
-    y_pred = clf.predict(X)
-    assert len(y_pred) == len(y)
-    assert not np.isnan(y_pred).all()
+    pl = BinaryClassificationPipeline(component_graph, random_seed=5)
+    assert pl.random_seed == 5
+    assert pl.get_component("Random Forest").random_seed == 3
+    assert pl.get_component("Random Forest B").random_seed == 4
 
 
-@patch(
-    "evalml.pipelines.components.ensemble.SklearnStackedEnsembleClassifier._stacking_estimator_class"
-)
-def test_stacked_ensemble_does_not_overwrite_pipeline_random_seed(
-    mock_stack, logistic_regression_binary_pipeline_class
-):
-    input_pipelines = [
-        logistic_regression_binary_pipeline_class(parameters={}, random_seed=3),
-        logistic_regression_binary_pipeline_class(parameters={}, random_seed=4),
+def test_stacked_problem_types():
+    assert ProblemTypes.BINARY in StackedEnsembleClassifier.supported_problem_types
+    assert ProblemTypes.MULTICLASS in StackedEnsembleClassifier.supported_problem_types
+    assert StackedEnsembleClassifier.supported_problem_types == [
+        ProblemTypes.BINARY,
+        ProblemTypes.MULTICLASS,
+        ProblemTypes.TIME_SERIES_BINARY,
+        ProblemTypes.TIME_SERIES_MULTICLASS,
     ]
-    clf = SklearnStackedEnsembleClassifier(
-        input_pipelines=input_pipelines, random_seed=5, n_jobs=1
-    )
-    estimators_used_in_ensemble = mock_stack.call_args[1]["estimators"]
-    assert clf.random_seed == 5
-    assert estimators_used_in_ensemble[0][1].pipeline.random_seed == 3
-    assert estimators_used_in_ensemble[1][1].pipeline.random_seed == 4
-
-
-def test_stacked_ensemble_multilevel(logistic_regression_binary_pipeline_class):
-    # checks passing a stacked ensemble classifier as a final estimator
-    X = pd.DataFrame(np.random.rand(50, 5))
-    y = pd.Series([1, 0] * 25)
-    base = SklearnStackedEnsembleClassifier(
-        input_pipelines=[logistic_regression_binary_pipeline_class(parameters={})],
-        n_jobs=1,
+
+
+def test_stacked_ensemble_nondefault_y(X_y_binary):
+    pytest.importorskip(
+        "imblearn.over_sampling",
+        reason="Skipping nondefault y test because imblearn not installed",
     )
-    clf = SklearnStackedEnsembleClassifier(
-        input_pipelines=[logistic_regression_binary_pipeline_class(parameters={})],
-        final_estimator=base,
-        n_jobs=1,
+    X, y = X_y_binary
+    input_pipelines = [
```

Comment:
Interesting... how does `fit()` work here? Doesn't the first pipeline's oversampler cause the two input pipelines to have different number of rows for the final estimator?

Reply:
When we had this bug previously (i.e. when we used to just use the input y blindly for the metalearner rather than take from an oversampler), we would get a mismatched input data error for this estimator. The `fit()` makes sure that it'll fit the pipeline without raising a RuntimeError and the `predict()` ensures that when we run that function, the input size match out the expected output size.

Reply:
Yeah, I'm curious how we're able to fit when the two input pipelines produce different number of predictions? I just want to clarify what's going on for my own sake 😂 

Using this example, let's say our dataset is 100 rows. The first input pipeline has a sampler that will oversample to 125 rows, while the second pipeline does not have any sampler. 

Is this not an issue because these two input pipelines are already fitted so the sampler does not modify the data, but instead we just call `predict` using the two input pipeline's estimators to create data for the metalearner? Is that why this isn't an issue? 🤔

Reply:
Chatted with @angela97lin about this. We decided to just update this test case for now so both pipelines have an oversampler and to make detecting pipelines with a different amount of rows into an estimator a separate issue.